internal/cri/server/images/image_pull.go

```diff
@@ -97,12 +97,65 @@ import (
 // after we pull the image? How to manage the disk usage of contents? If some
 // contents are missing but snapshots are ready, is the image still "READY"?
 
+// private context key for sandbox id propagation
+type ctxKey string
+
+const ctxKeySandboxID ctxKey = "cri.sandbox.id"
+
 // PullImage pulls an image with authentication config.
 func (c *GRPCCRIImageService) PullImage(ctx context.Context, r *runtime.PullImageRequest) (_ *runtime.PullImageResponse, err error) {
-
+	ctx, rpcSpan := tracing.StartSpan(ctx, tracing.Name("cri.image", "grpc_pull"))
+	defer func() {
+		if err != nil {
+			rpcSpan.SetStatus(err)
+		}
+		rpcSpan.End()
+	}()
 	imageRef := r.GetImage().GetImage()
 
+	// Extract sandbox id best-effort from annotations.
+	// There is no SandboxId field in PullImageRequest in CRI v1.
+	var sandboxID string
+	if r.SandboxConfig != nil && r.SandboxConfig.Annotations != nil {
+		// Common fallback keys that may carry sandbox id.
+		if v, ok := r.SandboxConfig.Annotations["io.kubernetes.cri.sandbox-id"]; ok && v != "" {
+			sandboxID = v
+		} else if v, ok := r.SandboxConfig.Annotations["k8s.v1.cni.cncf.io/sandbox-id"]; ok && v != "" {
```

Comment:
??

Reply:
thx don't get me wrong one day we might do an OCI/CNCF pod/sandbox specification .. just not yet.. not completely surprised the container networking folks (cni) had an annotation given the default network namespace being attached to all containers in the sandbox.